sdk/analytics/src/main/java/com/klaviyo/analytics/state/StateSideEffects.kt

```diff
@@ -110,7 +116,29 @@ internal class StateSideEffects(
     }
 
     private fun afterApiRequest(request: ApiRequest) = when {
-        request is PushTokenApiRequest && request.status == KlaviyoApiRequest.Status.Failed -> {
+        request.responseCode == HTTP_BAD_REQUEST -> {
+            request.responseBody?.let { responseBody ->
+                val error = KlaviyoErrorResponseDecoder.fromJson(JSONObject(responseBody))
+                error.errors.find { it.title == ErrorResponse.INVALID_INPUT_TITLE }?.let { inputError ->
+                    when (inputError.source?.pointer) {
+                        KlaviyoErrorSource.EMAIL_PATH -> {
+                            (Registry.get<State>() as? KlaviyoState)?.resetEmail()
+                            Registry.log.warning("Invalid email - resetting email state to null")
+                        }
+                        KlaviyoErrorSource.PHONE_NUMBER_PATH -> {
+                            (Registry.get<State>() as? KlaviyoState)?.resetPhoneNumber()
+                            Registry.log.warning(
+                                "Invalid phone number - resetting phone number state to null"
+                            )
+                        }
+                        else -> {
+                            Registry.log.warning("Input error: ${inputError.detail}")
```

Comment:
Another potential debate - do we wan to directly expose backend responses like this to the logs? I actually think here it wouldn't be a bad idea, since we do format it nicely for the other issues and I think it would be helpful for context. If we are cool with this I might just log the inputError.detail for the above cases as well, since they add more details than I did above

Reply:
Yeah there's nothing private about that, I think we also log the whole API response on error elsewhere.